src/Lecture4.hs

```diff
@@ -254,7 +353,7 @@ the file doesn't have any products.
 -}
 
 calculateStats :: String -> String
-calculateStats = error "TODO"
+calculateStats ls = displayStats . combineRows . fromList . mapMaybe parseRow $ lines ls
```

Comment:
`fromList` is an unsafe function. For example, after `mapMaybe` you may get an empty list and `fromList` will throw an error